src/nibpkg/read.nim

```diff
@@ -9,11 +9,11 @@ type Read* = object
 
 proc process_read*(s: string, idx: SvIndex, k: int = 25, spacedSeeds: bool = false, space: int = 50): Read =
     # find SVs with kmers intersecting with those from this read.
-    var kmers = Dna(s).dna_to_kmers(k)
+    var kmers = (cast[Dna](s)).dna_to_kmers(k)
```

Comment:
Hmmm. I guess you're right. `Dna(s)` could perform a copy. Not sure. But `cast[DNA](s)` is fast.